### PR TITLE
Add domain type as gremlinSource generic, fix #161.

### LINKED ISSUE #161: Generic of recoverDomain and GremlinSource

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

**Comment by Incarnation-p-lee** [comment]
@fabiencoppens 
thanks for comments.
ci failure due to Travis network issue，exist ID is expected for testing exception. As I am in National festival, and will go through the pr later.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/source/AbstractGremlinSource.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public abstract class AbstractGremlinSource implements GremlinSource {
+public abstract class AbstractGremlinSource<T> implements GremlinSource<T> {
```
[the remark this answers is not in the document]
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/source/AbstractGremlinSource.java]: should be like this, I attempt to remove switch code inf GremlinEntityType there, so leave this alone here.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java [hunk 2]
```diff
@@ -12,46 +12,44 @@
...
         if (!(source instanceof GremlinSourceGraph)) {
             throw new GremlinUnexpectedSourceTypeException("should be the instance of GremlinSourceGraph");
         }
 
         final List<String> scriptList = new ArrayList<>();
         final GremlinSourceGraph sourceGraph = (GremlinSourceGraph) source;
-        final GremlinScriptLiteralVertex scriptVertex = new GremlinScriptLiteralVertex();
-        final GremlinScriptLiteralEdge scriptEdge = new GremlinScriptLiteralEdge();
+        final List<GremlinSource> vertexes = (List<GremlinSource>) sourceGraph.getVertexSet();
+        final List<GremlinSource> edges = (List<GremlinSource>) sourceGraph.getEdgeSet();
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java]: Why the type cast for List is required?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java [hunk 3]
```diff
@@ -60,19 +58,20 @@
...
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java]: Same as above.

Comment (sophiaso) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java]: Seems generateUpdateScript and generateInsertScript are almost the same? Can these two methods be combined to handle 2 different(or more) cases?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralGraph.java]: The GremlinSource is generic, if use raw type of GremlinSource, the List<GremlinSource> will be erased to Object.

**Pull request merged**